spec.md

```diff
@@ -1848,8 +1854,12 @@ message NodePublishVolumeRequest {
   // The path to which the volume will be published. It MUST be an
   // absolute path in the root filesystem of the process serving this
   // request. The CO SHALL ensure uniqueness of target_path per volume.
-  // The CO SHALL ensure that the path exists, and that the process
-  // serving the request has `read` and `write` permissions to the path.
+  // The SP MUST NOT use symbolic link to publish the volume. The CO
```

Comment:
The use of symlink is naturally incompatible with the need to create empty file/directory (i.e., mount points) for volumes because SP then needs to delete the mount points created by the CO, and replace it with a symlink. The same argument applies to the use of `mknod` for block volumes.

I am actually leaning toward just dictating that it is has to be mount based. I think this align with our goal for SP to write one plugin that works for all COs.

What do you guys think? cc @saad-ali @julian-hj ?

Reply:
Do we also stop people from having part of PATH being symbolic link? I remember - some users define `/var/lib/kubelet` to be a symbolic link for example and in that case the path will be `/var/lib/kubelet/plugins/volumes/csi/xxx`.

Reply:
We switched to not letting CO to create the mount point. SP can do that. CO needs to make sure SP has the permission to do that.